iceoryx_binding_c/test/moduletests/test_wait_set.cpp

```diff
@@ -422,6 +422,25 @@ TIMING_TEST_F(iox_ws_test, WaitIsBlockingTillTriggered, Repeat(5), [&] {
     TIMING_TEST_EXPECT_TRUE(waitWasCalled.load());
 });
 
+TIMING_TEST_F(iox_ws_test, WaitIsNonBlockingAfterMarkForDestruction, Repeat(5), [&] {
+    std::atomic_bool waitWasCalled{false};
+    std::thread t([&] {
+        iox_ws_wait(m_sut, NULL, 0U, &m_missedElements);
+        iox_ws_wait(m_sut, NULL, 0U, &m_missedElements);
+        iox_ws_wait(m_sut, NULL, 0U, &m_missedElements);
+        waitWasCalled.store(true);
+    });
+
+    std::this_thread::sleep_for(std::chrono::milliseconds(100));
+    TIMING_TEST_EXPECT_FALSE(waitWasCalled.load());
+
+    iox_ws_mark_for_destruction(m_sut);
```

Comment:
So this is used to wake the waitset up without a trigger (which I requested). Personally I think destruction is not the only reason why we would want to do this (i.e. I want to wake up but continue working), so the function is not named correctly/the use case too narrow.

Reply:
But this was the easiest implementation since the WaitSet is not usable anymore after you called this - the wait and timed wait are not blocking anymore and are always returning an empty vector.